test/conformance/behaviors/sig-node/pod-spec.yaml

```diff
@@ -1,43 +1,47 @@
   description: When a Pod resource is created with a single container and sufficient resources, a Pod MUST be created on a node with the specified container image.
-- id: pod/basic-delete
+- id: pod/spec/basic-delete
   description: When a Pod resource is delete, the Pod containers must receive a TERM signal and the Pod MUST be deleted.
-- id: pod/hostname
+- id: pod/spec/hostname
   description: When the hostname field is set, a container running in the Pod MUST report the hostname as the specified value.
-- id: pod/subdomain
+- id: pod/spec/subdomain
   description: If specified, the fully qualified Pod hostname will be "<hostname>.<subdomain>.<pod
     namespace>.svc.<cluster domain>". If not specified, the pod will not have a
     domainname at all.
-- id: pod/terminationGracePeriodSeconds/in-spec
+- id: pod/spec/terminationGracePeriodSeconds/in-spec
   description: When the terminationGracePeriodSeconds is specified in the spec,
     processes running in the Pod MUST NOT receive a hard termination signal for at
     least that number of seconds after a delete request.
-- id: pod/terminationGracePeriodSeconds/in-delete
+- id: pod/spec/terminationGracePeriodSeconds/in-delete
   description: When the terminationGracePeriodSeconds is specified in a delete request,
     processes running in the Pod MUST NOT receive a hard termination signal for at
     least that number of seconds after the delete request.
-- id: pod/activeDeadlineSeconds
+- id: pod/spec/activeDeadlineSeconds
   description: Optional duration in seconds the pod may be active on the node relative
     to StartTime before the system will actively try to mark it failed and kill
     associated containers. Value must be a positive integer.
-- id: pod/hostNetwork/true
+- id: pod/spec/hostNetwork/true
   description: When hostNetwork is set to true, the Pod MUST use the host's network
     namespace.
-- id: pod/hostNetwork/false
+- id: pod/spec/hostNetwork/false
   description: When hostNetwork is set to false, the Pod MUST NOT use the host's network
     namespace.
-- id: pod/hostPID/true
+- id: pod/spec/hostPID/true
   description: When hostPID is set to true, the Pod MUST use the host's process
     namespace.
-- id: pod/hostPID/false
+- id: pod/spec/hostPID/false
   description: When hostPID is set to false, the Pod MUST NOT use the host's process
     namespace.
-- id: pod/hostIPC/true
+- id: pod/spec/hostIPC/true
   description: When hostIPC is set to true, the Pod MUST use the host's inter-process
     communication namespace.
-- id: pod/hostIPC/false
+- id: pod/spec/hostIPC/false
   description: When hostIPC is set to false, the Pod MUST NOT use the host's inter-process
     communication namespace.
+- id: pod/spec/basic-update
+  decription: Create a Pod with a unique label. Query for the Pod with the label as selector MUST be successful. Update the pod to change the value of the Label. Query for the Pod with the new value for the label MUST be successful.
```

Comment:
This isn't a useful behavior description. It should be much smaller in scope. In fact this is represents several behaviors: creating a pod with a label, querying by label, and changing a label. Those all fall into "basic machinery" of the API server. As I recall we need to have a discussion with sig api-machinery about a way to generically test these behaviors across all resources. This would apply to anything utilizing only metadata.

For this particular test, which already exists, we can add those few behaviors. Something like:
- When creating a Pod you MUST be able to set a label, and querying using Pods using a namespace and label selector for that label MUST return all Pods with that label in that namespace and no other Pods. [ yes this is in fact multiple behaviors but so tightly bound I am ok with it being one ]
- A patch request MUST be able to update a label on a Pod. Querying for the Pod with the new label in that namespace and no other Pods. 

I do not think a "basic-update" is a useful behavior, at least not for podspec which has very few mutable fields. Instead, we should identify the specific fields that are mutable within a podspec, and create behaviors describing what happens when you mutate those fields. So, for example, tolerations on a pod are mutable. We should have behaviors describing how taints & tolerations work - there are some conformance tests around that already. The behaviors should capture the expression of the change, not just that it happened in the API. That is, it should capture the data plane changes.

Reply:
Not sure on patch vs put here. Are they both possible? do they both behave the same way?

Reply:
I don't have an opinion on patch vs put.

For the initial load of behaviours, I'd like to get a list through that we can start soon rather that wait for every.

Reply:
I do agree that the basic update behavior isn't useful since it applies to the "basic machinery". For now though, I'd like to avoid changing the tests when we're in the process of just porting over the behaviors to the new linkage machinery.

I divided this into two behaviors, but I agree this test will need to be revisited in the future.